docs/source/advanced/optimized_multi_gpu.rst

```diff
@@ -0,0 +1,661 @@
+Memory Optimized Multi-GPU Training
+===================================
+
+When training large models or fitting larger batch sizes on multi-gpu compute, Lightning provides advanced optimized multi-gpu plugins to support these cases.
+
+For example if you'd like to train a large billion parameter transformer model, or to scale your batch size when training a semi-supervised learning model, using a Lightning optimized distributed training plugin will offer substantial improvements
+in memory usage.
+
+Note that some of the extreme memory saving configurations will affect the speed of training. This Speed/Memory trade-off in most cases can be adjusted.
+
+Some of these memory efficient plugins rely on offloading onto other forms of memory, such as CPU RAM or NVMe. This means you can even see memory benefits on a **single GPU**, using a plugin such as :ref:`deepspeed-zero-stage-3-offload`.
+
+Choosing an Optimized Multi-GPU Plugin
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Currently all Memory Optimized Multi-GPU plugins shard the model states across your GPUs; just in different ways.
+
+This means as you scale up the number of GPUs, you can reach the number of model parameters you'd like to train.
+
```

Comment:
should we could also highlight DDP communication hooks?

Reply:
Yes!!! Great idea :)